Cmake fixes

michal-narajowski opened the pull request:
- cmake: Escape quotes in linker flags

- cmake: Fix undeterministic build error by sorting includes
The order of include directories was non-deterministic and caused issues with
including wrong files which resulted in 'undefined reference' errors.

utzig commented on newt/builder/cmake.go:
```diff
@@ -263,7 +244,7 @@ func CmakeCompilerInfoWrite(w io.Writer, archiveFile string, bpkg *BuildPackage,
 	includes = append(includes, otherIncludes...)
 
-	includes = removeDuplicates(includes)
+	includes = util.SortFields(includes...)
```
Is it not a problem to leave duplicates?

michal-narajowski replied:
This function also removes duplicates. It is used by newt when compiling, so it should be ok.

utzig replied:
Ah right, `SortFields` calls `UniqueStrings` but that is not documented behavior!

michal-narajowski commented:
I added a comment that explains what sortFields() does and added a commit that sorts and removes duplicate compilation flags to account for possible issues with ordering in this aspect.
